tasks/extract.js

```diff
@@ -0,0 +1,230 @@
+var path = require('path');
+var fs = require('fs');
+
+var acorn = require('acorn');
+var acornWalk = require('../node_modules/acorn/dist/walk');
+var moment = require('moment');
+var nunjucksParser = require('nunjucks').parser;
+var nunjucksNodes = require('../node_modules/nunjucks/src/nodes');
+
+var PO_HEADER = [
+  'msgid ""',
+  'msgstr ""',
+  '"Project-Id-Version: PACKAGE VERSION\\n"',
+  '"Report-Msgid-Bugs-To: \\n"',
+  '"POT-Creation-Date: ' + moment().format('%yyyy-%mm-%d %I:%M%Z') + '\\n"',
+  '"PO-Revision-Date: YEAR-MO-DA HO:MI+ZONE\\n"',
+  '"Last-Translator: Automatically generated\\n"',
+  '"Language-Team: none\\n"',
+  '"Language: \\n"',
+  '"MIME-Version: 1.0\\n"',
+  '"Content-Type: text/plain; charset=UTF-8\\n"',
+  '"Content-Transfer-Encoding: 8bit\\n"',
+  '"X-Generator: Translate Toolkit 1.6.0\\n"',
+  '"Plural-Forms: nplurals=2; plural=(n != 1);\\n"',
+].join('\n') + '\n\n'
+
+
+module.exports = function(grunt) {
+  var extensionToExtractors = {
+    '.html': extractNunjucks,
+    '.js': extractJavasript,
+  };
+
+  grunt.registerMultiTask('extract', 'Extract strings for l10n', function() {
+    this.files.map(function(file) {
+      var stringSets = file.src.map(function(filepath) {
+        var extension = path.extname(filepath)
+        var extractor = extensionToExtractors[extension];
+
+        if (extractor === undefined) {
+          grunt.fail.fatal('No extraction method defined for extension ' + extension + ' (' + filepath + ')');
+          return [];
+        }
+
+        return extractor(filepath);
+      });
+
+      // flatten stringSets
+      var extractedStrings = Array.prototype.concat.apply([], stringSets);
+      writePotFile(extractedStrings, file.dest);
+    })
+  });
+
+  function extractNunjucks(filepath) {
+    var contents = grunt.file.read(filepath);
+    try {
+      var parseTree = nunjucksParser.parse(contents);
+    } catch (err) {
+      console.log(JSON.stringify(err));
+      grunt.fail.warn('Error while parsing ' +
+                      filepath + ':' + err.lineno + ':' + err.colno +
+                      '\n' + err.toString());
+    }
+
+    return parseTree.findAll(nunjucksNodes.FunCall)
+    .filter(function(node) {
+      // Exclude functions calls that aren't to gettext.
+      return (
+        node.name &&
+        node.name instanceof nunjucksNodes.Symbol &&
+        ['_', '_plural', 'gettext', 'ngettext'].indexOf(node.name.value) !== -1
```

Comment:
Shouldn't this be `['_', '_plural']` only? From `l10n.js`, we're not adding `gettext` and `ngettext` to the `nunjucksEnv`.